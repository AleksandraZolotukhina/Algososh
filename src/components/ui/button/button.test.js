import { Button } from "./button";
import { fireEvent } from "@testing-library/react";
import { render, screen } from "@testing-library/react";

describe("Отрисовка кнопки", () => {
    it("Отрисовка кнопки с текстом", () => {
        render(<Button text={"Some text"} />);
        const button = screen.getByRole("button");
        expect(button).toMatchSnapshot();
    })
    it("Отрисовка кнопки без текста", () => {
        render(<Button />);
        const button = screen.getByRole("button");
        expect(button).toMatchSnapshot();
    })
    it("Отрисовка заблокированной кнопки", () => {
        render(<Button disabled={true} />);

        const button = screen.getByRole("button");
        expect(button).toMatchSnapshot();
    })
    it("Отрисовка кнопки с индикацией загрузки", () => {
        render(<Button isLoader={true} />);
        const button = screen.getByRole("button");
        expect(button).toMatchSnapshot();
    })
    it("Корректность вызова колбека при клике на кнопку", () => {
        const func = jest.fn();
        render(<Button onClick={func} />);
        const button = screen.getByRole("button");
        fireEvent.click(button);
        expect(func).toHaveBeenCalled();
    })
})
