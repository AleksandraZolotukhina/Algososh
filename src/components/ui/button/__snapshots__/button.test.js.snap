// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Отрисовка кнопки Отрисовка заблокированной кнопки 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  disabled=""
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Отрисовка кнопки Отрисовка кнопки без текста 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Отрисовка кнопки Отрисовка кнопки с индикацией загрузки 1`] = `
<button
  class="text text_type_button text_color_primary button undefined loader "
  disabled=""
  type="button"
>
  <img
    alt="Загрузка."
    class="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Отрисовка кнопки Отрисовка кнопки с текстом 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  >
    Some text
  </p>
</button>
`;
